File: LeetCode/[E] - 349. Intersection of Two Arrays/solution.py
from typing import List


class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1.sort()
        nums2.sort()
        l, r = 0, 0
        lenNums1, lenNums2 = len(nums1), len(nums2)
        used = set()
        res = []
        while l < lenNums1 and r < lenNums2:
            if nums1[l] == nums2[r]:
                if nums1[l] not in used:
                    res.append(nums1[l])
                    used.add(nums1[l])
                l += 1
                r += 1
            elif nums1[l] < nums2[r]:
                l += 1
            else:
                r += 1
        return res

File: LeetCode/[E] - 349. Intersection of Two Arrays/solution.go
package main

func intersection(nums1 []int, nums2 []int) []int {
	m1, m2 := make(map[int]struct{}), make(map[int]struct{})
	for _, n1 := range nums1 {
		m1[n1] = struct{}{}
	}
	for _, n2 := range nums2 {
		m2[n2] = struct{}{}
	}
	var res []int
	for k := range m1 {
		if _, ok := m2[k]; ok {
			res = append(res, k)
		}
	}
	return res
}
